### ISSUE TITLE
1 Blind not responding in HomeKit, other 6 do perfectly.  Will respond in MySmartBlind app

### ISSUE THREAD

**Issue by randymassey** [apexad/homebridge-mysmartblinds-bridge#17]
**Bug or issue information**
I have this set up, & 6 of 7 blinds are working pretty much flawlessly, both with Siri in HomeKit via this plugin & HOOBS (nice plugin BTW) & in the MySmartBlinds app.

All of the units are new (so it possible there is a faulty unit, but since the MSB app will work, I veer towards no).  This unit was also wiped & re-paired.  All MSB firmware for hub & blinds is current.  (My background as FYI is out of IT, & beta testing further back, at software development companies.  So, I am familiar & comfortable in these configs, etc. just as FYI)  I have a L & R blind in my office & the L just times out, even after re-pairing it.

The log will show it pushed the command & what happens is the blue ring on the MSB Hub will end up just spinning & spinning.  The log in HOOBS will "think" it succeeded, as the commands will issue correctly in the log, but then timeout in the Home app in iOS.  If I issue commands to all of the other blinds together or individually, no problems. I also reboot HOOBS, routers, bridges, hubs etc, relative to any re-pairing to MSB motor head units & re-testing, I also power cycle the MSB hub.  I will paste the log below, but interestingly it will show the command & think it has performed it, but really the HUB's blue ring spins until it times out.  But if I immediately go into the MSB iOS app, & move that blind, it will react immediately, as it should, so the hub seems to be communicating with the blind in question ok, in the official app.

In the log example pasted below, everything seems to THINK it's working, but the one blind will timeout in HomeKit, & basically not respond, or sometimes HomeKit will think it's followed the command & say is now closed, when really it never reacted & the blind is still open (in its last state, but as an example) & any automations including this blind, will also fail if this blind is included.  No errors seem to come back into the log.  

I can send more of the log of needed, just LMK, this log paste was just after a HOOBS reboot, & then trying an automation to send to all of the blinds.

**Homebridge setup information**
HOOBS v3.2.6 on a Pi4.
Running 3 plugin's only, this one, the one for the BOND controller (v2.7.4) & the one for SleepIQ (v3.2.10).  All of these items work / respond / do not time out, & if I leave this ONE blind out of the mix, all of the existing blinds will function without timeouts.  Stumped...  ;)

**Logs**
```bash
7/17/2020, 4:35:31 PM [MySmartBlindsBridge] Initializing platform accessory "Dining Room Right"...
7/17/2020, 4:35:31 PM [MySmartBlindsBridge] Initializing platform accessory "Living Room Side Window"...
7/17/2020, 4:35:31 PM [MySmartBlindsBridge] Initializing platform accessory "Living Room Front"...
7/17/2020, 4:35:31 PM [MySmartBlindsBridge] Initializing platform accessory "Office Right Office"...
7/17/2020, 4:35:31 PM [MySmartBlindsBridge] Initializing platform accessory "Dining Room Left"...
7/17/2020, 4:35:31 PM [MySmartBlindsBridge] Initializing platform accessory "Office Left Office"...
7/17/2020, 4:35:31 PM [MySmartBlindsBridge] Initializing platform accessory "Kitchen Kitchen Sink"...
7/17/2020, 4:35:31 PM Bridge is running on port 51826.
7/17/2020, 4:36:23 PM [MySmartBlindsBridge] Dining Room Right setTargetPosition from -1 to 45
7/17/2020, 4:36:23 PM [MySmartBlindsBridge] Dining Room Left setTargetPosition from -1 to 24
7/17/2020, 4:36:23 PM [MySmartBlindsBridge] Kitchen Kitchen Sink setTargetPosition from -1 to 69
7/17/2020, 4:36:23 PM [MySmartBlindsBridge] Living Room Side Window setTargetPosition from -1 to 50
7/17/2020, 4:36:23 PM [MySmartBlindsBridge] Living Room Front setTargetPosition from -1 to 100
7/17/2020, 4:36:23 PM [MySmartBlindsBridge] Office Right Office setTargetPosition from -1 to 0
7/17/2020, 4:36:23 PM [MySmartBlindsBridge] Office Left Office setTargetPosition from -1 to 0
7/17/2020, 4:36:48 PM [MySmartBlindsBridge] Dining Room Left currentPosition is now 24
7/17/2020, 4:36:48 PM [MySmartBlindsBridge] Living Room Front currentPosition is now 100
7/17/2020, 4:36:48 PM [MySmartBlindsBridge] Living Room Side Window currentPosition is now 50
7/17/2020, 4:36:48 PM [MySmartBlindsBridge] Dining Room Right currentPosition is now 45
7/17/2020, 4:36:48 PM [MySmartBlindsBridge] Office Left Office currentPosition is now 0
7/17/2020, 4:36:48 PM [MySmartBlindsBridge] Kitchen Kitchen Sink currentPosition is now 69
7/17/2020, 4:36:48 PM [MySmartBlindsBridge] Office Right Office currentPosition is now 0
7/17/2020, 4:37:20 PM [MySmartBlindsBridge] Dining Room Right setTargetPosition from 45 to 45
7/17/2020, 4:37:20 PM [MySmartBlindsBridge] Dining Room Left setTargetPosition from 24 to 24
7/17/2020, 4:37:20 PM [MySmartBlindsBridge] Kitchen Kitchen Sink setTargetPosition from 69 to 69
7/17/2020, 4:37:20 PM [MySmartBlindsBridge] Living Room Side Window setTargetPosition from 50 to 50
7/17/2020, 4:37:20 PM [MySmartBlindsBridge] Living Room Front setTargetPosition from 100 to 100
7/17/2020, 4:37:20 PM [MySmartBlindsBridge] Office Right Office setTargetPosition from 0 to 0
7/17/2020, 4:37:20 PM [MySmartBlindsBridge] Office Left Office setTargetPosition from 0 to 0
7/17/2020, 4:37:46 PM [MySmartBlindsBridge] Living Room Side Window currentPosition is now 50
7/17/2020, 4:37:46 PM [MySmartBlindsBridge] Living Room Front currentPosition is now 100
7/17/2020, 4:37:46 PM [MySmartBlindsBridge] Office Left Office currentPosition is now 0
7/17/2020, 4:37:46 PM [MySmartBlindsBridge] Office Right Office currentPosition is now 0
7/17/2020, 4:37:46 PM [MySmartBlindsBridge] Dining Room Left currentPosition is now 24
7/17/2020, 4:37:46 PM [MySmartBlindsBridge] Kitchen Kitchen Sink currentPosition is now 69
7/17/2020, 4:37:46 PM [MySmartBlindsBridge] Dining Room Right currentPosition is now 45
```

**Comment by apexad** [comment]
Can you turn on the `allow debug` option and get some additional logs that hopefully shows an error or something?

**Comment by apexad** [comment]
Closing due to inactivity.
Will likely be resolved with the plugin rewrite mentioned in https://github.com/apexad/homebridge-mysmartblinds-bridge/issues/13 (and I'm hoping to finally work on this very soon).

**Comment by apexad** [comment]
Just checking in here.  The plugin has been converted to a dynamic platform and it is my hope that this issue is resolved. If not please re-open this issue or create another one.

**Comment by randymassey** [comment]
Yes, hi there, sorry meant to follow this up, but have had some medical diversions of late...  

I did end up just waiting for your changes, once I knew you were working on these changes in that future, coming revision.  (I did have a problem with another plugin that derailed me a fair bit as well, as it was actually taking down HOOBS & ultimately & we can not figure out how, but also several of my most recently added actual HomeKit items were vanishing too, I had to leave that plugin out of the the mix).  

Anyway, that gave me an opportunity to wipe HOOBS, reinstall & start from scratch, & then test this scenario again against your update, as it timed to be about the same time window as when you released the revision, one tries every which way to avoid HOOBS rebuilds, but eventually LOL, it a good thing...  So, it's actually been working really well now & none of the issues discussed above are happening now under this revision of your plugin.  I have 10 blind conversion units in my house in total & they are all working quite smoothly in truth.  Great work there sir...  Really good plugin.  😏

RandyM
